### PR TITLE
AppleMusicRichPresence: fix formatting when listening to radio

### PR DESCRIPTION
**Pull request by lumap**
You can listen to radio stations through the Apple Music app, such as the BBC or France Inter. These return... unusual values as the current activity. This PR hides the irrelevant/missing info

### CODE REVIEW

File: src/plugins/appleMusic.desktop/index.tsx [hunk 1]
```diff
@@ -243,17 +245,17 @@ export default definePlugin({
 
             name: customFormat(settings.store.nameString, trackData),
             details: customFormat(settings.store.detailsString, trackData),
-            state: customFormat(settings.store.stateString, trackData),
+            state: isARadioStation ? undefined : customFormat(settings.store.stateString, trackData),
 
-            timestamps: (settings.store.enableTimestamps ? {
+            timestamps: (trackData.playerPosition && trackData.duration) ? (settings.store.enableTimestamps ? {
```
Comment (ryanccn) [inline src/plugins/appleMusic.desktop/index.tsx:250]: Why the nested ternary? Putting all the conditions together is functionally equivalent. Also, `playerPosition` being 0 would be a perfectly valid value to put in timestamps.

```suggestion
            timestamps: (Number.isNaN(trackData.duration) && settings.store.enableTimestamps ? {
```

File: src/plugins/appleMusic.desktop/index.tsx [hunk 2]
```diff
@@ -243,17 +245,17 @@ export default definePlugin({
...
                 start: Date.now() - (trackData.playerPosition * 1000),
                 end: Date.now() - (trackData.playerPosition * 1000) + (trackData.duration * 1000),
-            } : undefined),
+            } : undefined) : undefined,
```
Comment (ryanccn) [inline src/plugins/appleMusic.desktop/index.tsx:253]:
```suggestion
            } : undefined),
```

File: src/plugins/appleMusic.desktop/index.tsx [hunk 3]
```diff
@@ -243,17 +245,17 @@ export default definePlugin({
...
 
             assets,
 
-            buttons: buttons.length ? buttons.map(v => v.label) : undefined,
-            metadata: { button_urls: buttons.map(v => v.url) || undefined, },
+            buttons: isARadioStation ? undefined : buttons.length ? buttons.map(v => v.label) : undefined,
+            metadata: isARadioStation ? undefined : { button_urls: buttons.map(v => v.url) || undefined, },
```
Comment (ryanccn) [inline src/plugins/appleMusic.desktop/index.tsx:258]: Unnecessary nested ternaries here as well; merging them is possible.

File: src/plugins/appleMusic.desktop/index.tsx [hunk 4]
```diff
@@ -212,14 +212,16 @@ export default definePlugin({
 
         const assets: ActivityAssets = {};
 
+        const isARadioStation = (trackData.album === "missing value") && Number.isNaN(trackData.duration) && !trackData.artist && (trackData.playerPosition === 0); // this is the values returned by Apple if you're listening to a radio station, e.g. BBC
```
[the remark this answers is not in the document]
Comment (lumap) [reply src/plugins/appleMusic.desktop/index.tsx]: > Would it be possible to cut down on the amount of conditions? I feel like two or three of these (e.g. Number.isNaN(trackData.duration) && trackData.playerPosition === 0) would be reasonably enough.
I'm not sure, Apple Music may return some of these values in other situations, e.g. if the track is not fully loaded. I did this long check for safety, but if you think some can be removed, please tell me which!

[the remark this answers is not in the document]
Comment (ryanccn) [reply src/plugins/appleMusic.desktop/index.tsx]: I think that `Number.isNaN(trackData.duration) && trackData.playerPosition === 0` would theoretically suffice, but in the end you should test things and figure out if it actually works as it should

File: src/plugins/appleMusic.desktop/index.tsx [hunk 5]
```diff
@@ -243,17 +245,17 @@ export default definePlugin({
...
-            state: customFormat(settings.store.stateString, trackData),
+            state: isRadio ? undefined : customFormat(settings.store.stateString, trackData),
 
-            timestamps: (settings.store.enableTimestamps ? {
-                start: Date.now() - (trackData.playerPosition * 1000),
-                end: Date.now() - (trackData.playerPosition * 1000) + (trackData.duration * 1000),
-            } : undefined),
+            timestamps: (trackData.playerPosition && trackData.duration && settings.store.enableTimestamps) ? {
+            start: Date.now() - (trackData.playerPosition * 1000),
+            end: Date.now() - (trackData.playerPosition * 1000) + (trackData.duration * 1000),
```
Comment (ryanccn) [inline src/plugins/appleMusic.desktop/index.tsx:252]: Odd indentation changes. Have you run lint?

File: src/plugins/appleMusic.desktop/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (lumap) [reply src/plugins/appleMusic.desktop/index.tsx]: yeah it works well enough from my testing. committed it

**Comment by Vendicated** [comment]
thanks

**Pull request merged**